shared/pkg/config/runtime_scanner.go

```diff
@@ -40,19 +49,131 @@ func setRegistryConfigDefaults() {
 func LoadRuntimeScannerRegistryConfig(imageID string) *Registry {
 	setRegistryConfigDefaults()
 
-	username, password, err := creds.ExtractCredentials(imageID)
-	if err != nil {
-		log.Warnf("Failed to extract credentials for imageID %v: %v", imageID, err)
+	var auth Auth
```

Comment:
so we are not using `ExtractCredentials` any more? we are not using the `DefaultKeychain` that uses the docker config file?
Can't we use a single docker config file with multiple creds in it?

Reply:
The job doesn't have a docker config file outside of the ones mounted in by the secret, so I dropped ExtractCredentials and instead copied the logic that kubelet/go-containerregistry does here: https://github.com/google/go-containerregistry/blob/main/pkg/authn/kubernetes/keychain.go#L63-L134 except instead of loading the secrets from kubernetes I'm loading them from disk.

Reply:
ok, make sense. so we should delete `ExtractCredentials` and all code related.

Reply:
I added a commit which removes the dead code including ExtractCredentials 👍